block/local/adapter.go

```diff
@@ -127,7 +135,27 @@ func (l *Adapter) Remove(obj block.ObjectPointer) error {
 		return err
 	}
 	p = filepath.Clean(p)
-	return os.Remove(p)
+	err = os.Remove(p)
+	if err != nil {
+		return err
+	}
+	if l.removeEmptyDir {
+		dir := filepath.Dir(p)
+		removeEmptyDir(dir)
+	}
+	return nil
+}
+
+func removeEmptyDir(dir string) {
+	d, err := os.Open(dir)
+	if err != nil {
+		return
+	}
+	_, err = d.Readdir(1)
+	_ = d.Close()
+	if err == io.EOF {
+		_ = os.Remove(dir)
+	}
```

Comment:
It's usually best to try something on a filesystem rather than to try to reproduce the OS internal logic.  In this case, I think the current code fails when the user has execute but not read permissions on directory.
Perhaps something like this (untested)?
```suggestion
	err := os.Remove(dir)
	if err != nil && !errors.Is(err, syscall.ENOTEMPTY)) {
		return err
	}
	return nil
```

Reply:
cool, so if we also do not report errors related to our empty folder cleanup I just call

```go
_ = os.Remove(dir)
```

Reply:
Yup!  But I'd rather still log the error before ignoring it, information always helps debug issues.  Of course if you did that you'd still need the `if` above :-/